integration-tests/src/test/java/org/apache/fineract/integrationtests/SavingsAccountTransactionsIntegrationTest.java

```diff
@@ -0,0 +1,379 @@
+
+        GetSavingsAccountTransactionsResponse transactionsResponse = this.savingsAccountHelper
+                .getSavingsTransactionsWithQueryParams(savingsId, queryParametersMap);
+        Assertions.assertNotNull(transactionsResponse);
+        assertEquals(3, transactionsResponse.getTotalFilteredRecords());
+        Assertions.assertNotNull(transactionsResponse.getPageItems());
+        assertEquals(3, transactionsResponse.getPageItems().size());
+        Set<GetSavingsAccountTransactionsPageItem> pageItems = transactionsResponse.getPageItems();
+        List<BigDecimal> actualList = new ArrayList<>();
+        pageItems.forEach(data -> {
+            actualList.add(data.getAmount());
+        });
+        boolean isAscending = isListOrdered(actualList, null);
+        // Default Sort is ASC if sortOrder is not provided
+        assertEquals(true, isAscending);
+
+    }
+
+    @Test
+    public void testSavingsTransactionsSortByTransactionDateAsc() {
+        final String clientCreateDate = "25 Apr 2023";
+        final String startDate1 = "01 May 2023";
+        final String startDate2 = "04 May 2023";
+        final String startDate3 = "07 May 2023";
+        final Integer clientID = ClientHelper.createClient(this.requestSpec, this.responseSpec, clientCreateDate);
+        Assertions.assertNotNull(clientID);
+        Map<String, String> queryParametersMap = new HashMap<>();
+        queryParametersMap.put("orderBy", "dateOf");
+        queryParametersMap.put("sortOrder", "ASC");
+        final Integer savingsId = createSavingsAccountDailyPosting(clientID, clientCreateDate);
+
+        this.savingsAccountHelper.depositToSavingsAccount(savingsId, "1000", startDate1, CommonConstants.RESPONSE_RESOURCE_ID);
+        this.savingsAccountHelper.depositToSavingsAccount(savingsId, "500", startDate2, CommonConstants.RESPONSE_RESOURCE_ID);
+        this.savingsAccountHelper.depositToSavingsAccount(savingsId, "700", startDate3, CommonConstants.RESPONSE_RESOURCE_ID);
+
+        GetSavingsAccountTransactionsResponse transactionsResponse = this.savingsAccountHelper
+                .getSavingsTransactionsWithQueryParams(savingsId, queryParametersMap);
+        Assertions.assertNotNull(transactionsResponse);
+        assertEquals(3, transactionsResponse.getTotalFilteredRecords());
+        Assertions.assertNotNull(transactionsResponse.getPageItems());
+        assertEquals(3, transactionsResponse.getPageItems().size());
+        Set<GetSavingsAccountTransactionsPageItem> pageItems = transactionsResponse.getPageItems();
+        List<LocalDate> actualList = new ArrayList<>();
+        pageItems.forEach(data -> {
+            actualList.add(data.getDate());
+        });
+        boolean isAscending = isDateListOrdered(actualList, "ASC");
+        assertEquals(true, isAscending);
+
+    }
+
+    @Test
+    public void testSavingsTransactionsFilterByTransactionType() {
```

Comment:
This test is just half of the pie because you never tested whether without the filtering applied, the post interest transaction is also returned; i.e. you don't know if your filter parameter that you passed has been handled or it's because by default the API filters out non deposit transactions.